damnit/backend/db.py

```diff
@@ -2,46 +2,98 @@
 import logging
 import sqlite3
 from collections.abc import MutableMapping, ValuesView, ItemsView
+from dataclasses import dataclass, asdict
+from datetime import datetime, timezone
 from pathlib import Path
 from secrets import token_hex
+from typing import Any
 
 from ..context import UserEditableVariable
+from ..ctxsupport.ctxrunner import DataType, get_variable_type
 
-DB_NAME = 'runs.sqlite'
+DB_NAME = Path('runs.sqlite')
 
 log = logging.getLogger(__name__)
 
-# More columns can be added to runs() table depending on the data
-BASE_SCHEMA = """
-CREATE TABLE IF NOT EXISTS runs(proposal, runnr, start_time, added_at, comment);
-CREATE UNIQUE INDEX IF NOT EXISTS proposal_run ON runs (proposal, runnr);
+V1_SCHEMA = """
+CREATE TABLE IF NOT EXISTS run_info(proposal, run, start_time, added_at);
+CREATE UNIQUE INDEX IF NOT EXISTS proposal_run ON run_info (proposal, run);
+
+CREATE TABLE IF NOT EXISTS run_variables(proposal, run, name, version, value, timestamp, stored_type, max_diff, provenance);
```

Comment:
The `stored_type` column seems somewhat confusing to me, because we're storing it with the reduced data, but it describes the un-reduced data (where those are different, e.g. arrays.

Not sure what to do about this, but I want to remember to think about it.